tools/PI/DevHome.PI/Helpers/ExternalTool.cs

```diff
@@ -75,50 +78,39 @@ public string OtherArgs
         ArgPrefix = argprefix;
         OtherArgs = otherArgs;
         IsPinned = isPinned;
+        PinGlyph = IsPinned ? CommonHelper.UnpinGlyph : CommonHelper.PinGlyph;
 
         ID = Guid.NewGuid().ToString();
 
         if (!string.IsNullOrEmpty(executable))
         {
-            GetToolImage();
-            GetMenuIcon();
+            GetIcons();
         }
     }
 
-    private async void GetToolImage()
+    partial void OnIsPinnedChanged(bool oldValue, bool newValue)

```

Comment:
What does `oldValue` and `newValue`` represent?

Reply:
For 

 [ObservableProperty]
 private bool _isPinned;

It's what the old value of _isPinned was, and what the new value is when it gets changed.